data/data/bootstrap/baremetal/files/usr/local/bin/master-bmh-update.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+. /usr/local/bin/release-image.sh
+
+wait_for_existence() {
+    while [ ! -e "${1}" ]
+    do
+        sleep 5
+    done
+}
+
+# Wait for the API to become active
+until curl -s http://localhost:6385/v1/nodes; do
+    echo Waiting for API
+    sleep 20
+done
+
+# Wait for the nodes to become active after introspection.
+while curl -s http://localhost:6385/v1/nodes | jq '.nodes[] | .provision_state' | grep -v active; do
+    echo Waiting for nodes to become active
+    sleep 20
+done
+
+echo Nodes are all active
+echo Waiting for bootkube to finish
+
+# Wait for bootkube to come up.  We want to talk to the full instance not
+# the bootstrap openshift. This one only sleeps for 5 as we are getting
+# closer to being ready.
+wait_for_existence /opt/openshift/.bootkube.done
```

Comment:
What do you think about this instead?

I don't think we need to wait for bootkube.done, we can modify the CR's before we pivot to the cluster.


```bash
export KUBECONFIG=/etc/kubernetes/kubeconfig

until oc get baremetalhosts -n openshift-machine-api; do
   echo Waiting for BareMetalHosts to appear...
done
```

Reply:
OK cool.  In the latest version now.. trying it out